lib/node_modules/@stdlib/string/base/replace-before/README.md

```diff
@@ -0,0 +1,233 @@
+limitations under the License.
+
+-->
+
+# replaceBefore
+
+> Replace the substring before the first occurrence of a specified search string.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var replaceBefore = require( '@stdlib/string/base/replace-before' );
+```
+
+#### replaceBefore( str, search, replacement )
+
+Replaces the substring before the first occurrence of a specified search string.
+
+```javascript
+var str = 'beep boop';
+var replacement = 'loop';
+var out = replaceBefore( str, ' ', replacement );
+// returns 'loop boop'
+
+replacement = 'bar';
+out = replaceBefore( str, 'o' , replacement );
+// returns 'baroop'
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+-   If a substring is not present in a provided string, the function returns the input string.
+-   If provided an empty substring, the function returns an empty string.
+
+</section>
+
+<!-- /.notes -->
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var substringBefore = require( '@stdlib/string/base/replace-before' );
+
+var out = substringBefore( 'beep boop', 'p', 'see' );
+// returns 'seep boop'
+
+out = substringBefore( 'Hello World!', 'xyz', 'foo' );
+// returns 'fooHello World!'
+
+out = substringBefore( 'Hello World!', '', 'foo' );
+// returns 'fooHello World!'
+
+out = substringBefore( '', 'xyz' , 'foo');
+// returns 'foo'
+```
+
+</section>
+
+<!-- /.examples -->
+
+<!-- Section for describing a command-line interface. -->
+
+* * *
+
+<section class="cli">
+
+## CLI
+
+<!-- CLI usage documentation. -->
+
+<section class="usage">
+
+### Usage
+
+```text
+Usage: replace-before [options] --search=<string> [<string>] --replacement=<string> [<string>]
+
+Options:
+
+  -h,    --help                Print this message.
+  -V,    --version             Print the package version.
+         --search string       Search string.
+         --split sep           Delimiter for stdin data. Default: '/\\r?\\n/'.
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- CLI usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+### Notes
+
+-   If the split separator is a [regular expression][mdn-regexp], ensure that the `split` option is either properly escaped or enclosed in quotes.
+
+    ```bash
+    # Not escaped...
+    $ echo -n $'foo\nbar\nbaz' | replace-before --search a --split /\r?\n/
+
+    # Escaped...
+    $ echo -n $'foo\nbar\nbaz' | replace-before --search a --split /\\r?\\n/
```

Comment:
Based on L118, the `replacement` option is required. Yet, this example does not include it in either L142 or L145.